### DISCUSSION TITLE
作者大大，能不能出一个自建邮箱服务器的项目

### DISCUSSION THREAD

**Discussion by ghost** [dreamhunter2333/cloudflare_temp_email#422] [General]
作者大大，能不能出一个SMTP+IMAP+POP3+反垃圾邮件+防病毒网络管理+电子邮件在cf workers上的项目，就是把cf当成自行搭建的邮箱服务器，可以通过网页端和这些协议收发邮件，不再是这种临时邮箱，而是自建邮局。

**Comment by dreamhunter2333** [comment]
> 作者大大，能不能出一个SMTP+IMAP+POP3+反垃圾邮件+防病毒网络管理+电子邮件在cf workers上的项目，就是把cf当成自行搭建的邮箱服务器，可以通过网页端和这些协议收发邮件，不再是这种临时邮箱，而是自建邮局。


目前配合网页的admin控制台 和 smtp imap proxy 使用可以达到你说的自建邮局，proxy 暂时没适配所有的客户端，使用 thundrbird 应该没大问题

反垃圾邮件+防病毒网络管理 这两个没太懂是什么功能，dns 里可以配置一些拒绝来源邮件的规则，

对于协议相关的，cf 无法提供 非 http 或者 websocket 的连接，

目前IMAP这类协议只能通过自建 proxy 的方式实现

对于发送邮件，可以直接使用 resend 的 smtp服务，就可以绕过自建这个步骤，

如果要自建邮局的话，目前推荐这个轻量服务 https://github.com/Jinnrry/PMail 占用不大，缺点就是需要一个允许smtp发送的服务器
